packages/cli/openapi-parser/src/openapi/v3/AbstractOpenAPIV3ParserContext.ts

```diff
@@ -46,38 +46,86 @@ export abstract class AbstractOpenAPIV3ParserContext implements SchemaParserCont
     }
 
     public resolveSchemaReference(schema: OpenAPIV3.ReferenceObject): OpenAPIV3.SchemaObject {
+        const startsWithSchemaReferencePrefix = schema.$ref.startsWith(SCHEMA_REFERENCE_PREFIX);
+        const endsWithSchemaSuffix = schema.$ref.endsWith("/schema");
```

Comment:
If we think about these cases, they aren't actually that different -- 

In one case you point to `#/components/schemas/MySchema` and in another case you might point to `#/components/schemas/MySchema/application~json/200/schema`

Reply:
Feels like we should just update the code to take the reference, convert it into an array of keys `string[]` and then index into the OpenAPI spec accordingly. At any point if we hit undefined, we should throw an error.

Reply:
Updated in second attempt